contracts/dapis/DapiServer.sol

```diff
@@ -645,362 +581,265 @@ contract DapiServer is
+                    values[ind] = beacon.value;
+                    accumulatedTimestamp += beacon.timestamp;
+                }
             }
+            require(signatureCount == 0, "Signature count not met");
+            bytes32 beaconSetId = deriveBeaconSetId(beaconIds);
+            uint32 updatedTimestamp = uint32(
+                accumulatedTimestamp / beaconCount
+            );
+            require(
+                updatedTimestamp >
+                    oevProxyToIdToDataFeed[oevProxy][beaconSetId].timestamp,
+                "Does not update timestamp"
+            );
+            int224 updatedValue = int224(median(values));
+            oevProxyToIdToDataFeed[oevProxy][beaconSetId] = DataFeed({
+                value: updatedValue,
+                timestamp: updatedTimestamp
+            });
+            oevProxyToBalance[oevProxy] += msg.value;
+            emit UpdatedOevProxyBeaconSetWithSignedData(
+                beaconSetId,
+                oevProxy,
+                updatedValue,
+                updatedTimestamp
+            );
+        } else if (beaconCount == 1) {
+            (
+                bytes32 beaconId,
+                int224 beaconValue,
+                uint32 beaconTimestamp
+            ) = decodeOevSignedData(bidMetadataHash, signedData[0]);
+            require(beaconTimestamp != 0, "Missing data");
+            require(
+                beaconTimestamp >
+                    oevProxyToIdToDataFeed[oevProxy][beaconId].timestamp,
+                "Does not update timestamp"
+            );
+            oevProxyToIdToDataFeed[oevProxy][beaconId] = DataFeed({
+                value: beaconValue,
+                timestamp: beaconTimestamp
+            });
+            oevProxyToBalance[oevProxy] += msg.value;
+            emit UpdatedOevProxyBeaconWithSignedData(
+                beaconId,
+                oevProxy,
+                beaconValue,
+                beaconTimestamp
+            );
+        } else {
+            revert("Specified no Beacons");
         }
-        beaconSetId = deriveBeaconSetId(beaconIds);
-        uint32 updatedTimestamp = uint32(accumulatedTimestamp / beaconCount);
-        require(
-            updatedTimestamp >=
-                oevProxyToIdToDataFeed[msg.sender][beaconSetId].timestamp,
-            "Updated value outdated"
-        );
-        int224 updatedValue = int224(median(values));
-        oevProxyToIdToDataFeed[msg.sender][beaconSetId] = DataFeed({
-            value: updatedValue,
-            timestamp: updatedTimestamp
-        });
-        emit UpdatedOevProxyBeaconSetWithSignedData(
-            beaconSetId,
-            msg.sender,
-            updatedValue,
-            updatedTimestamp
-        );
+    }
+
+    /// @notice Withdraws the balance of the OEV proxy to the respective
+    /// beneficiary account
+    /// @param oevProxy OEV proxy
+    function withdraw(address oevProxy) external override {
```

Comment:
Let me double check I got this right:
1. anyone can trigger the transfer
2. the balance to be transferred is the sum of all the bid amounts the winning searcher has sent when doing updates
3. the oevBeneficiary is set in the proxy constructor with an address, this address is probably set by us in the factory contract. right? These are API3 proceeds for granting the update to a searcher.

Reply:
1. Yes (though I accept that this is ugly). The reasoning is that the beneficiary may be something like a DAO treasury and may not be able to trigger the withdrawal itself. We also don't want to forward the transfer to the beneficiary in case the beneficiary default `payable` function is heavy (pull payments > push payments).
2. It's the sum of all paid bids to do updates for a specific proxy (the funds may come from different searchers). Funds are collected per proxy and not per beneficiary because the beneficiary may want to implement a proxy-specific incentivization scheme (for example there is a LP pool for each asset and the OEV collected from each asset is awarded to the respective pool -- though it can be argued that the beneficiary addresses will be different then).
3. The project that will deploy the proxy specifies `oevBeneficiary`. They enter that in a textbox on the Market and click deploy, we have them deploy their proxy with this address. Then, they can withdraw the OEV to the address whenever they want.